### PR TITLE
fix(capabilities): fix checking window/showDocument client capability

### PR DESCRIPTION
**Pull request by williamboman**
Per the [spec](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#window_showDocument), the `window.showDocument` is an object: `{ support: boolean }`. This currently causes this LSP server to think that Neovim's LSP client supports `window/showDocument`, when it doesn't.

### CODE REVIEW

File: src/svlangserver.ts [hunk 1]
```diff
@@ -95,9 +95,9 @@ let svhiercalculator: SystemVerilogHierarchyCalculator = new SystemVerilogHierar
 let settings = default_settings;
 
 connection.onInitialize((params: InitializeParams) => {
-    hasConfigurationCapability = !!(params.capabilities.workspace && !!params.capabilities.workspace.configuration);
-    hasShowWindowCapability = !!(params.capabilities.window && params.capabilities.window.showDocument);
-    clientName = !!params.clientInfo ? params.clientInfo.name : undefined;
+    hasConfigurationCapability = params.capabilities.workspace?.configuration == true;
+    hasShowWindowCapability = params.capabilities.window?.showDocument?.support == true;
+    clientName = params.clientInfo?.name;
```
Comment (williamboman) [inline src/svlangserver.ts:100]: Some other stylistic changes, using the [optional chaining operator `?.`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining), happy to revert the unrelated changes though

**Comment by kkanhere** [comment]
Thanks!
